hydro_deploy/core/src/deployment.rs

```diff
@@ -41,6 +41,16 @@ impl Deployment {
         self.add_service(|id| CustomService::new(id, on, external_ports))
     }
 
+    // TODO(mingwei): generalize to `run_until(fut: impl Future)`?
```

Comment:
I like this idea of generalizing.

Reply:
I'll make this an issue because I'm not sure about the exact mechanics of
```
self.deploy().await?;
self.start().await?;
// ..
self.stop().await?;
```
in terms of RAII cleanup -- do we want to be able to stop the run during deployment?